src/ParallelAlgorithms/LinearSolver/Schwarz/ElementCenteredSubdomainData.hpp

```diff
@@ -0,0 +1,214 @@
+ *
+ * An element-centered subdomain consists of a central element and overlap
+ * regions with all neighboring elements. This class holds data on such a
+ * subdomain. It supports vector space operations (addition and scalar
+ * multiplication) and an inner product, which allow to use this data type with
+ * linear solvers (see e.g. `LinearSolver::Serial::Gmres`).
+ */
+template <size_t Dim, typename TagsList>
+struct ElementCenteredSubdomainData {
+  static constexpr size_t volume_dim = Dim;
+  using ElementData = Variables<TagsList>;
+  using OverlapData = ElementData;
+
+  ElementCenteredSubdomainData() = default;
+  ElementCenteredSubdomainData(const ElementCenteredSubdomainData&) = default;
+  ElementCenteredSubdomainData& operator=(
+      const ElementCenteredSubdomainData&) noexcept = default;
+  ElementCenteredSubdomainData(ElementCenteredSubdomainData&&) noexcept =
+      default;
+  ElementCenteredSubdomainData& operator=(
+      ElementCenteredSubdomainData&&) noexcept = default;
+  ~ElementCenteredSubdomainData() noexcept = default;
+
+  explicit ElementCenteredSubdomainData(
+      const size_t element_num_points) noexcept
+      : element_data{element_num_points} {}
+
+  ElementCenteredSubdomainData(
+      Variables<TagsList> local_element_data,
+      OverlapMap<Dim, Variables<TagsList>> local_overlap_data) noexcept
+      : element_data{std::move(local_element_data)},
+        overlap_data{std::move(local_overlap_data)} {}
+
+  void pup(PUP::er& p) noexcept {  // NOLINT
+    p | element_data;
+    p | overlap_data;
+  }
+
+  template <typename RhsTagsList>
+  ElementCenteredSubdomainData& operator+=(
+      const ElementCenteredSubdomainData<Dim, RhsTagsList>& rhs) noexcept {
+    element_data += rhs.element_data;
+    for (auto& id_and_overlap_data : overlap_data) {
+      id_and_overlap_data.second +=
+          rhs.overlap_data.at(id_and_overlap_data.first);
+    }
+    return *this;
+  }
+
+  template <typename RhsTagsList>
+  ElementCenteredSubdomainData& operator-=(
+      const ElementCenteredSubdomainData<Dim, RhsTagsList>& rhs) noexcept {
+    element_data -= rhs.element_data;
+    for (auto& id_and_overlap_data : overlap_data) {
+      id_and_overlap_data.second -=
+          rhs.overlap_data.at(id_and_overlap_data.first);
+    }
+    return *this;
+  }
+
+  ElementCenteredSubdomainData& operator*=(const double scalar) noexcept {
+    element_data *= scalar;
+    for (auto& id_and_overlap_data : overlap_data) {
+      id_and_overlap_data.second *= scalar;
+    }
+    return *this;
+  }
+
+  ElementCenteredSubdomainData& operator/=(const double scalar) noexcept {
+    element_data /= scalar;
+    for (auto& id_and_overlap_data : overlap_data) {
+      id_and_overlap_data.second /= scalar;
+    }
+    return *this;
+  }
+
+  ElementData element_data{};
+  OverlapMap<Dim, OverlapData> overlap_data{};
+};
+
+template <size_t Dim, typename LhsTagsList, typename RhsTagsList>
+decltype(auto) operator+(
```

Comment:
The `Variables` class supports arithmetics with tag lists that differ by prefixes, so I'm just carrying over that functionality here. It can be useful when working with prefixed tags.